proposals/867-kubeflow-documentation-ai/README.md

```diff
@@ -0,0 +1,577 @@
+                  }
+              )
+            env:
+            - name: GITHUB_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: github-secret
+                  key: token
+          restartPolicy: OnFailure
+```
+
+**GitHub Token Secret**
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: github-secret
+type: Opaque
+stringData:
+  token: <your-github-personal-access-token>
+```
+
+**Enhanced RAG Search with PR Data**
+```python
+class EnhancedRAGSearcher(RAGSearcher):
+    def search_with_prs(self, query: str, include_prs: bool = True) -> List[Dict]:
+        """Search both documentation and PR data"""
+        
+        # Get regular documentation results
+        doc_results = self.search(query, limit=3)
+        
+        if not include_prs:
+            return doc_results
+        
+        # Search PR collections
+        pr_results = []
+        query_vector = self.encoder.encode(query).tolist()
+        
+        for collection in self.client.list_collections():
+            if collection.startswith("pr_"):
+                try:
+                    results = self.client.search(
+                        collection_name=collection,
+                        data=[query_vector],
+                        limit=2,
+                        output_fields=["repo_name", "title", "author", "state", "url", "content"]
+                    )
+                    
+                    for result in results[0]:
+                        if (1 - result['distance']) > 0.4:  # Threshold for PR relevance
+                            entity = result['entity']
+                            pr_results.append({
+                                'type': 'pull_request',
+                                'repo_name': entity['repo_name'],
+                                'title': entity['title'],
+                                'author': entity['author'],
+                                'state': entity['state'],
+                                'url': entity['url'],
+                                'content': entity['content'][:500],
+                                'score': 1 - result['distance']
+                            })
+                except Exception as e:
+                    continue
+        
+        # Combine and sort results
+        combined_results = doc_results + pr_results
+        combined_results.sort(key=lambda x: x['score'], reverse=True)
+        
+        return combined_results[:5]
+```
+
+### Implementation Plan
+
+#### Phase 1: Repository Setup and Infrastructure Planning
+- Create a new repository under the Kubeflow organization: `kubeflow/docs-bot`
+- Define cloud resource requirements for:
+  - LLM inference (GPU/CPU requirements for Llama 3.1 8B or Mistral 7B)
+  - Milvus vector database deployment
+  - Backend service hosting
+- Establish project structure with clear separation of concerns:
+  - `/milvus` - yaml files for deploying milvus
+  - `/config` - Cluster configuration
```

Comment:
Shall we store it in `manifests/` to be consistent with other Kubeflow projects:
https://github.com/kubeflow/trainer/tree/master/manifests
https://github.com/kubeflow/model-registry/tree/main/manifests